src/shogun/mathematics/Random.cpp

```diff
@@ -120,7 +121,36 @@ void CRandom::init()
 	m_sfmt_32 = SG_MALLOC(sfmt_t, 1);
 	m_sfmt_64 = SG_MALLOC(sfmt_t, 1);
 	m_dsfmt = SG_MALLOC(dsfmt_t, 1);
-	reinit(m_seed);
+	reinit();
+
+	if(!obj)
+		return;
+
+	obj->watch_method<bool>("reinit", [&]() {
+		this->reinit();
+		return true;
+	});
+
+	obj->watch_param("seed", &m_seed);
+
+	std::function<void(CSGObject*, uint32_t&)> callback;
+	callback = [&](CSGObject* object, uint32_t& seed) {
+		object->for_each_param_of_type<CSGObject*>(
+		    [&](const std::string& name, CSGObject** param) {
+			    if ((*param)->has("seed"))
+				{
+				    (*param)->put_quietly("seed", seed);
+				    (*param)->run("reinit");
```

Comment:
why can't we call the method directly here?

Reply:
SGObject doesn't have it, we're sure it's there since "seed" exists, which implies there is a Random mixin.

Reply:
but you know that the object implements the random mixin or?

Reply:
we know by implication: "seed" exists, since it must implement the Random mixin. maybe there is a better way.